packages/edit-site/src/components/template-switcher/index.js

```diff
@@ -67,49 +71,70 @@ export default function TemplateSwitcher( {
 		setThemePreviewVisible( () => false );
 	};
 
+	const [ homeId, setHomeId ] = useState();
+	useEffect( () => {
+		const effect = async () => {
+			try {
+				const { success, data } = await fetch(
+					addQueryArgs( '/', { '_wp-find-template': true } )
+				).then( ( res ) => res.json() );
+				if ( success ) {
+					let newHomeId = data.ID;
+					if ( newHomeId === null ) {
+						const { getEntityRecords } = resolveSelect( 'core' );
+						newHomeId = getEntityRecords(
```

Comment:
You need to await the promise that this returns.

Reply:
Oops :sweat_smile: These chained network fetches :sweat_smile: 
Fixed in 215e984